microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileValidator.java

```diff
@@ -161,6 +178,32 @@ private void validatePropertyValue(String propertyName, ItemMetadata metadata, P
 		}
 	}
 
+	/**
+	 * Validates the property value expressions (${other.property}) of the given property.
+	 * 
+	 * Checks if the property expression is closed, and if the referenced property exists.
+	 * 
+	 * @param property The property to validate
+	 * @param allProperties A list of all the properties defined in the property file and in the project info
+	 */
+	private void validatePropertyValueExpressions(Property property, Set<String> allProperties) {
+		if (property.getValue() == null) {
+			return;
+		}
+		for (Node child : property.getValue().getChildren()) {
+			if (child != null && child.getNodeType() == NodeType.PROPERTY_VALUE_EXPRESSION) {
+				PropertyValueExpression propValExpr = (PropertyValueExpression) child;
+				String refdProp = propValExpr.getReferencedPropertyName();
+				if (!propValExpr.isClosed()) {
+					addDiagnostic("Missing '}'", propValExpr, DiagnosticSeverity.Error, ValidationType.missingCloseBrace.name());

```

Comment:
I wasn't sure whether it would be better to use `ValidationType.syntax` or to make a new `ValidationType`.

Reply:
Use ValidationType.syntax (see my comments about settings).